zrml/global-disputes/src/migrations.rs

```diff
@@ -0,0 +1,383 @@
+// Copyright 2022-2023 Forecasting Technologies LTD.
+// Copyright 2021-2022 Zeitgeist PM LLC.
+//
+// This file is part of Zeitgeist.
+//
+// Zeitgeist is free software: you can redistribute it and/or modify it
+// under the terms of the GNU General Public License as published by the
+// Free Software Foundation, either version 3 of the License, or (at
+// your option) any later version.
+//
+// Zeitgeist is distributed in the hope that it will be useful, but
+// WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the GNU
+// General Public License for more details.
+//
+// You should have received a copy of the GNU General Public License
+// along with Zeitgeist. If not, see <https://www.gnu.org/licenses/>.
+
+extern crate alloc;
+
+use crate::{types::*, Config, Pallet as GDPallet, *};
+#[cfg(feature = "try-runtime")]
+use alloc::vec::Vec;
+use frame_support::{
+    dispatch::Weight,
+    log,
+    pallet_prelude::PhantomData,
+    traits::{Get, OnRuntimeUpgrade, StorageVersion},
+};
+use sp_runtime::traits::Saturating;
+
+#[cfg(feature = "try-runtime")]
+use alloc::collections::BTreeMap;
+#[cfg(feature = "try-runtime")]
+use parity_scale_codec::{Decode, Encode};
+#[cfg(feature = "try-runtime")]
+use scale_info::prelude::format;
+
+const GD_REQUIRED_STORAGE_VERSION: u16 = 0;
+const GD_NEXT_STORAGE_VERSION: u16 = 1;
+
+pub struct ModifyGlobalDisputesStructures<T>(PhantomData<T>);
+
+impl<T: Config + zrml_market_commons::Config> OnRuntimeUpgrade
+    for ModifyGlobalDisputesStructures<T>
+{
+    fn on_runtime_upgrade() -> Weight
+    where
+        T: Config,
+    {
+        let mut total_weight = T::DbWeight::get().reads(1);
+        let gd_version = StorageVersion::get::<GDPallet<T>>();
```

Comment:
`STORAGE_VERSION` was previously undefined, not sure if it returns `0` in that case (probably it does).

Reply:
https://paritytech.github.io/substrate/master/src/frame_support/traits/metadata.rs.html#219

get_or_default smells like using 0 if not defined